exchanges/request/retry_test.go

```diff
@@ -0,0 +1,124 @@
+package request_test
+
+import (
+	"net"
+	"net/http"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+)
+
+func TestDefaultRetryPolicy(t *testing.T) {
+	type args struct {
+		Error    error
+		Response *http.Response
+	}
+	type want struct {
+		Error error
+		Retry bool
+	}
+	testTable := map[string]struct {
+		Args args
+		Want want
+	}{
+		"DNS Error": {
+			Args: args{Error: &net.DNSError{Err: "fake"}},
+			Want: want{Error: &net.DNSError{Err: "fake"}},
+		},
+		"DNS Timeout": {
+			Args: args{Error: &net.DNSError{Err: "fake", IsTimeout: true}},
+			Want: want{Retry: true},
+		},
+		"Too Many Requests": {
+			Args: args{Response: &http.Response{StatusCode: http.StatusTooManyRequests}},
+			Want: want{Retry: true},
+		},
+		"Not Found": {
+			Args: args{Response: &http.Response{StatusCode: http.StatusNotFound}},
+		},
+		"Retry After": {
+			Args: args{Response: &http.Response{StatusCode: http.StatusTeapot, Header: http.Header{"Retry-After": []string{"0.5"}}}},
+			Want: want{Retry: true},
+		},
+	}
+
+	for name, tt := range testTable {
+		tt := tt
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			retry, err := request.DefaultRetryPolicy(tt.Args.Response, tt.Args.Error)
+
+			if exp := tt.Want.Error; exp != nil {
+				if !reflect.DeepEqual(err, exp) {
+					t.Fatalf("unexpected error\nexp: %#v, got: %#v", exp, err)
+				}
+				return
+			}
+
+			if err != nil {
+				t.Fatalf("unexpected error\nexp: <nil>, got: %#v", err)
+			}
+
+			if tt.Want.Retry != retry {
+				t.Fatalf("incorrect retry flag\nexp: %v, got: %v", tt.Want.Retry, retry)
+			}
+		})
+	}
+}
+
+func TestRetryAfter(t *testing.T) {
+	now := time.Date(2020, time.April, 20, 13, 31, 13, 0, time.UTC)
+
+	type args struct {
+		Now      time.Time
+		Response *http.Response
+	}
+	type want struct {
+		Delay time.Duration
+	}
+	testTable := map[string]struct {
+		Args args
+		Want want
+	}{
+		"No Response": {},
```

Comment:
I've omitted values here where the _zero value_ is being used, since they are populated by default.

If it is preferred, I can declare them explicitly, e.g:

```
"No Response": {
	Args:  args{Response: nil},
	Want: want{Delay: 0},
},
```

Setting `Now` is optional as it is only used in the case where the `Retry-After` header is present AND uses a date.

Reply:
Am happy to keep them omitted! This is an aim also for the project as a whole and makes the code neater